### DISCUSSION TITLE
No onBlur validation if field is empty

### DISCUSSION THREAD

**Discussion by jamiehaywood** [react-hook-form/react-hook-form#2397] [Q&A]
When I set up a form with the following:

```js
  const { register, handleSubmit, errors } = useForm<Form>({
    mode: 'all',
    reValidateMode: 'onBlur',
  });
```
```jsx
<TextInput
    errorMessage={errors?.email?.message}
    id="email"
    inputRef={register({required: "This field is required" })}
    label="Email"
    name="email"
    showError={!!errors?.email}
/>
```

When tabbing through the form _**but not inputting**_ I don't want the validation to fire onBlur. Is there an easy way to do this?

**Comment by bluebill1049** [comment]
<img width="686" alt="Screen Shot 2020-07-28 at 9 30 18 pm" src="https://user-images.githubusercontent.com/10513364/88660146-8ab92b00-d119-11ea-82a2-90cec7864c75.png">

u want `onChange` combine with `touched` for now.

**Comment by bluebill1049** [comment]
FYI: https://github.com/react-hook-form/react-hook-form/pull/2377 this feature is WIP, probably what you are after.
